hazelcast/src/main/java/com/hazelcast/nio/ascii/TextReadHandler.java

```diff
@@ -62,7 +62,7 @@
 
     private static final Map<String, CommandParser> MAP_COMMAND_PARSERS = new HashMap<String, CommandParser>();
 
-    private static final int CAPACITY = 500;
+    private static final int CAPACITY = 5000;
```

Comment:
What I means is, since `TextReadHandler` is single threaded, `ByteBuffer` can be replaced with a bigger one when needed (for example, create with double capacity, copy contents and destroy the old one). Still, a max-threshold will be needed to avoid creating very large buffers.

Or the whole logic can be changed to use a relatively small (1k, 4k whatever) buffer to build command incrementally. I am not sure if it's  really needed to hold all data read from socket in a single large buffer.